### ISSUE TITLE
LB service names in haproxy config

### ISSUE THREAD

**Issue by kaos** [rancher/rancher#6292]
Not sure what this will look like in the refactored LB code from #2179. But in case it still applies, I toss this out so it may be addressed, if needed/desired.

**Rancher Version:**

1.1.2

**Docker Version:**

1.12.2

**OS and where are the hosts located? (cloud, bare metal, etc):**

RHEL 7, bare metal

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**

Single node, external DB

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**

Cattle.

**Steps to Reproduce:**

Add LB, listen to a couple of ports, add some routes. Enable haproxy stats, load the stats page.

**Results:** 

Something like:

![image](https://cloud.githubusercontent.com/assets/72965/19429512/611fa9f4-9450-11e6-9efb-9bc92c07029f.png)

**Whish:**

Would love it if I could see which service it was from the haproxy stats page, rather than having to go about being a detective to map the id strings to services.. it's so tedious I haven't even attempted it.. ;)

**Comment by alena1108** [comment]
@kaos in refactored LB, you can assign a name for your backend ("backendName" flag on the API). We can't automatically derive it from the service/stack name as potentially the backend can consist of instances of multiple services.

**Comment by kaos** [comment]
@alena1108 Excellent. :)
Will it also be available to set this from the `rancher-compose` file?

**Comment by alena1108** [comment]
@kaos yes

**Comment by sangeethah** [comment]
Tested with latest build from master:

We are able to provide a name for a backend using portRules("backendName")  in LbConfig parameter when creating a service and  able to set specific configs to this backend by referring to this backend name  in config parameter.

Create LB with the following portRules set in lbConfig and  config parameters:

```
    port_rules = []
    port_rule = {"hostname": "www.abc1.com",
                 "path": "/service1.html",
                 "serviceId": 0,
                 "sourcePort": port,
                 "targetPort": "80",
                 "protocol": "http",
                 "backendName": "myrule1"
                 }
    port_rules.append(port_rule)
    port_rule = {"hostname": "www.abc2.com",
                 "path": "/service2.html",
                 "serviceId": 1,
                 "sourcePort": port,
                 "targetPort": "80",
                 "protocol": "http"
                 }
    port_rules.append(port_rule)

    config = "backend myrule1\nbalance first\n"
```

hapoxy config has the backend programmed with "balance first" as expected and accessing Lb ip for this backend respects the balance first configuration set.

```
listen default
bind *:42

frontend 1923
bind *:1923
mode http
acl 1923_www_abc2_com_service2_html_host hdr(host) -i www.abc2.com
acl 1923_www_abc2_com_service2_html_host hdr(host) -i www.abc2.com:1923
acl 1923_www_abc2_com_service2_html_path path_beg -i /service2.html
use_backend 1923_www_abc2_com_service2_html if 1923_www_abc2_com_service2_html_h
ost 1923_www_abc2_com_service2_html_path
acl myrule1_host hdr(host) -i www.abc1.com
acl myrule1_host hdr(host) -i www.abc1.com:1923
acl myrule1_path path_beg -i /service1.html
use_backend myrule1 if myrule1_host myrule1_path

backend 1923_www_abc2_com_service2_html
acl forwarded_proto hdr_cnt(X-Forwarded-Proto) eq 0
acl forwarded_port hdr_cnt(X-Forwarded-Port) eq 0
    http-request add-header X-Forwarded-Port %[dst_port] if forwarded_port
    http-request add-header X-Forwarded-Proto https if { ssl_fc } forwarded_prot
o
mode http
server 10.42.156.171 10.42.156.171:80
server 10.42.156.110 10.42.156.110:80

backend myrule1
acl forwarded_proto hdr_cnt(X-Forwarded-Proto) eq 0
acl forwarded_port hdr_cnt(X-Forwarded-Port) eq 0
    balance first
    http-request add-header X-Forwarded-Port %[dst_port] if forwarded_port
    http-request add-header X-Forwarded-Proto https if { ssl_fc } forwarded_prot
o
mode http
server 10.42.76.236 10.42.76.236:80
server 10.42.255.2 10.42.255.2:80
```

@kaos This feature will be available in 1.2 release.

**Comment by kaos** [comment]
Looks promising! :D

tors 3 nov. 2016 kl. 04:18 skrev Sangeetha Hariharah <
notifications@github.com>:
